modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -730,14 +733,40 @@ private void attachAnnotations(AnnotatableNode annotatableNode, int count) {
         tempAnnotAttachments.forEach(annot -> annotatableNode.addAnnotationAttachment(annot));
     }
 
-    public void addAssignmentStatement(boolean isVarDeclaration) {
+    public void addAssignmentStatement(DiagnosticPos pos, boolean isVarDeclaration) {
         ExpressionNode rExprNode = exprNodeStack.pop();
         List<ExpressionNode> lExprList = exprNodeListStack.pop();
-        AssignmentNode assignmentNode = TreeBuilder.createAssignmentNode();
+        BLangAssignment assignmentNode = (BLangAssignment) TreeBuilder.createAssignmentNode();
         assignmentNode.setExpression(rExprNode);
         assignmentNode.setDeclaredWithVar(isVarDeclaration);
+        assignmentNode.pos = pos;
         lExprList.forEach(expressionNode -> assignmentNode.addVariable((BLangVariableReference) expressionNode));
-        this.blockNodeStack.peek().addStatement(assignmentNode);
+        addStmtToCurrentBlock(assignmentNode);
+    }
+
+    public void startWhileStmt() {
+        startBlock();
+    }
+
+    public void addWhileStmt(DiagnosticPos pos) {
+        WhileNode whileNode = TreeBuilder.createWhileNode();
+        whileNode.setCondition(exprNodeStack.pop());
```

Comment:
Should we set the position to the while node also?

Reply:
Added with fe9a3b977913ea6b159a2384660e07cdcb311296. Thanks.